beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -546,3 +567,40 @@ func (f *ForkChoice) SetGenesisTime(genesisTime uint64) {
 func (f *ForkChoice) SetOriginRoot(root [32]byte) {
 	f.store.originRoot = root
 }
+
+// CachedHeadRoot returns the last cached head root
+func (f *ForkChoice) CachedHeadRoot() [32]byte {
+	f.store.nodesLock.RLock()
+	defer f.store.nodesLock.RUnlock()
+	node := f.store.headNode
+	if node == nil {
+		return [32]byte{}
+	}
+	return f.store.headNode.root
+}
+
+// FinalizedPayloadBlockHash returns the hash of the payload at the finalized checkpoint
+func (f *ForkChoice) FinalizedPayloadBlockHash() [32]byte {
+	f.store.nodesLock.RLock()
+	defer f.store.nodesLock.RUnlock()
+	root := f.FinalizedCheckpoint().Root
+	node := f.store.nodeByRoot[root]
+	if node == nil {
```

Comment:
This check is to prevent a panic, but at runtime, if the finalized checkpoint is not in forkchoice we are doomed anyway.